### PR TITLE
planner, executor: support SQL `show pump/drainer status`

### PR DESCRIPTION
**Pull request by caohe**
### What problem does this PR solve? 
Impl #9201 

> 2. Support SQL `show pump status` and `show drainer status`

### What is changed and how it works?
Changes `executor` and `planbuilder`. 
Just like `binlogctl`, it gets pump and drainer status from PD.

### Check List 

Tests 

 - Manual test(start pump and drainer, then execute `show pump status` and `show drainer status` to check)

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by caohe** [comment]
The address of PD seems not correct. And how can I add unit tests?

**Comment by caohe** [comment]
@WangXiangUSTC PTAL

**Comment by WangXiangUSTC** [comment]
thanks @caohe

File: executor/show.go [hunk 1]
```diff
@@ -996,6 +1006,58 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
...
+	for _, n := range nodes {
+		e.appendRow([]interface{}{n.NodeID, n.Addr, n.State, n.MaxCommitTS, utils.TSOToRoughTime(n.UpdateTS)})
+	}
+
+	return nil
+}
+
+// fetchShowDrainerStatus gets status of all drainers and fill them into e.rows.
+func (e *ShowExec) fetchShowDrainerStatus() error {
+	registry, err := createRegistry(config.GetGlobalConfig().Path)
```
Comment (WangXiangUSTC) [inline executor/show.go]: need close the registry https://github.com/pingcap/tidb-tools/blob/c71b1effdf1b05b0db12b9d8b544d035b910b74d/tidb-binlog/node/registry.go#L31

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -1755,6 +1755,9 @@ func buildShowSchema(s *ast.ShowStmt, isView bool) (schema *expression.Schema) {
 		names = []string{"View", "Create View", "character_set_client", "collation_connection"}
 	case ast.ShowCreateDatabase:
 		names = []string{"Database", "Create Database"}
+	case ast.ShowDrainerStatus:
+		names = []string{"NodeID", "Address", "State", "Max_Commit_Ts", "Update_Time"}
+		ftypes = []byte{mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeLonglong, mysql.TypeDatetime}
```
Comment (WangXiangUSTC) [inline planner/core/planbuilder.go]: can't get the Update_Time value, please check it

File: executor/show.go [hunk 3]
```diff
@@ -996,6 +1006,58 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
 	return nil
 }
 
+// fetchShowPumpStatus gets status of all pumps and fill them into e.rows.
+func (e *ShowExec) fetchShowPumpStatus() error {
```
Comment (WangXiangUSTC) [inline executor/show.go]: we can combine function `fetchShowPumpStatus` and `fetchShowDrainerStatus`

File: executor/show.go (hunk 1 above)
Comment (caohe) [reply executor/show.go]: Done. PTAL

File: executor/show.go (hunk 3 above)
Comment (caohe) [reply executor/show.go]: Done.

File: executor/show.go [hunk 4]
```diff
@@ -996,6 +1006,43 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
...
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = registry.Close()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, n := range nodes {
+		e.appendRow([]interface{}{n.NodeID, n.Addr, n.State, n.MaxCommitTS, utils.TSOToRoughTime(n.UpdateTS).Format("2006-01-02 15:04:05")})
```
Comment (WangXiangUSTC) [inline executor/show.go]: how about use types.TimeFormat?

**Comment by WangXiangUSTC** [comment]
rest LGTM

File: executor/show.go (hunk 4 above)
Comment (caohe) [reply executor/show.go]: thx. I didn't know it before.

**Comment by tiancaiamao** [comment]
LGTM
Although those kinds of feature is better to be implemented as plugin in my opinion...
PTAL @jackysp

**Comment by caohe** [comment]
@zz-jason OK. @WangXiangUSTC and I have both tested it manually. But I don't know how to add unit tests to it. It seems that I should start/mock a pump/drainer in the test code? Or simply check the number of rows and columns in the result like #6785?

**Comment by zz-jason** [comment]
Both "mock a pump/drainer in the test code" and "simply check the number of rows and columns in the result" are acceptable to me

**Comment by caohe** [comment]
I have tried a few times to add unit test for this sql clause with the help of @WangXiangUSTC . But there are some difficulties in mocking pd. So I am sorry but may be not able to add unit test this time. I will add unit test when I find a better way.

**Pull request merged**